internal/walker/rewriter.go

```diff
@@ -10,8 +10,14 @@ import (
 )
 
 type NodeRewriteFunc func(node *restic.Node, path string) *restic.Node
+type SnapshotSizeRewriteFunc func() SnapshotSize
```

Comment:
I'd prefer calling this function `QueryRewrittenSizeFunc` or similar. Currently, it sounds like it actually rewrites the snapshot's size, but that isn't the case.